pulsar-client-cpp/python/pulsar/__init__.py

```diff
@@ -288,7 +288,8 @@ def __init__(self, service_url,
                  log_conf_file_path=None,
                  use_tls=False,
                  tls_trust_certs_file_path=None,
-                 tls_allow_insecure_connection=False
+                 tls_allow_insecure_connection=False,
+                 tls_validate_hostname=True,
```

Comment:
@ivankelly  In Java and C++, we currently have the hostname validation turned off by default.

Reply:
We should probably change it to on by default, though it depends on what we expect deployments to look like. If users are accessing through a load balancer backed by proxies, it makes sense to validate the hostname. If connecting directly to brokers, maybe not, though a wildcard cert could still be used.